### PR TITLE
Add unsigned literals, and update suffixes

### PR DESCRIPTION
**Pull request by segeljakt**
Do not merge, need to fix tests first

### CODE REVIEW

**Comment by segeljakt** [comment]
Merging should now be OK.

File: src/main/scala/se/kth/cda/arc/ast/AST.scala [hunk 1]
```diff
@@ -14,6 +14,7 @@ sealed trait ASTNode {
 
 object AST {
 
+
```
Comment (frej) [inline src/main/scala/se/kth/cda/arc/ast/AST.scala]: unrelated white-space change?

Comment (segeljakt) [reply src/main/scala/se/kth/cda/arc/ast/AST.scala]: oops

Comment (segeljakt) [reply src/main/scala/se/kth/cda/arc/ast/AST.scala]: it is now undone 😄

**Comment by segeljakt** [comment]
Tests are passing now, @frej could you take a look if there is anything to change?

File: .gitignore [hunk 2]
```diff
@@ -133,3 +133,6 @@ arc-mlir/.guessed-arc-mlir
 
 # Editor backup files
 *~
+build/
+CMakeCache.txt
+CMakeFiles/
```
Comment (frej) [inline .gitignore]: The only reason for `CMakeCache.txt` and `CMakeFiles/` to occur in your source directory is if you did something wrong, and that could break proper builds in `build/`. So I think it is better to only add `build/`, that way we'll notice that something is wrong.

**Comment by segeljakt** [comment]
Oops, pushed too many commits

**Comment by segeljakt** [comment]
@frej I split it into 3 commits:
* One that fixes the Lexer tests
* One that changes the postfixes and adds tests for them.
* One that adds unsigned literals and adds tests for them.
Does it work or do you want the tests in separate commits?

**Comment by frej** [comment]
Now the split of the work into logical commits which should pass their tests looks good!

Unfortunately c812f80b5a484905557e874b1416bc4eed8b0857 and 94db47868498e1a285465d3c4abf474161113f90 fails `sbt test` and because of that I cannot run `sbt asssemble` and check the mlir tests.

**Comment by frej** [comment]
How do we get travis to run the tests for all commits and not just the tip?

**Comment by segeljakt** [comment]
@frej :shipit:?

**Pull request merged**